### PR TITLE
Fixes #7459: Reporting is broken for several ncf generic methods because the new class_prefix is not computer without this.callers_promisers, and any conditions using class_prefix are incorrectly evaluated

### PR DESCRIPTION
**Pull request by jooooooon**
https://www.rudder-project.org/redmine/issues/7459

### CODE REVIEW

File: tests/style/30_generic_methods/all_generic_methods_should_use_intermediary_promisers_slist.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Check that all generic_methods use a classes => classes_generic_two body and not the plain classes_generic
+# This is necessary for the transition from old to new reporting style, based on multiple class names
+
+FILES_TO_CHECK=`find "${NCF_TREE}/30_generic_methods/" -name "*.cf"`
+NB_ERROR=0
+for f in ${FILES_TO_CHECK}
+do
+
+  NB_BUNDLE=$(egrep '[^#]*"class_prefix"\s+string\s+=>\s+canonify\(join\("_",\s+"this.callers_promisers"\)\);' $f | wc -l)
```
Comment (peckpeck) [inline tests/style/30_generic_methods/all_generic_methods_should_use_intermediary_promisers_slist.sh]: Couldn't it be made more resilient with a regex like: '[^#]*"class_prefix"\s+string\s+=>.*this.callers_promisers.*'

**Comment by jooooooon** [comment]
PR updated

**Comment by ncharles** [comment]
looks good to me, i cant review the bash part however

**Comment by peckpeck** [comment]
Good, merging

**Pull request merged**